### DISCUSSION TITLE
Warning: multiple batches of metrics

### DISCUSSION THREAD

**Discussion by tchambers97** [artilleryio/artillery#1587] [Artillery]
Hello, any insight into what this warning means?
`Warning: multiple batches of metrics for period 1662051730000 2022-09-01T17:02:10.000Z`

**Comment by clearops** [comment]
we see this sometimes with load tests running with a lot of concurrency

if i understand correctly it means that messages from some of the artillery workers don't get processed within a certain time window (20-30 or seconds, not entirely sure). this has no effect on the number of requests actually sent to the api you're testing, but the metrics from that worker for that 10 second period won't be reflected in artillery's report

**Reply by tchambers97** [reply]
So, I'm seeing this happen continuously when running an artillery-playwright test in Bitbucket Pipelines. As a result, we're not getting any metrics back at all, other than VUs created. Is it possible that the metrics for every 10-second period are not being processed fast enough to be written? Is there any way to change how often we try to write metrics?

**Reply by tchambers97** [reply]
Launching workers... [18:34:02]
Waiting for Fargate... [18:34:03]
Waiting for workers to start: pending: 1 [18:34:33]
Waiting for workers to start: pending: 1 [18:34:43]
Waiting for workers to start: pending: 1 [18:34:53]
Waiting for workers to start: pending: 1 [18:35:03]
Waiting for workers to come online... [18:35:03]
Workers are running, waiting for reports... [18:36:44]
--------------------------------------
Metrics for period to: 18:37:00(+0000) (width: 0s)
--------------------------------------
vusers.created: ................................................................ 1
vusers.created_by_name.HomePage: ............................................... 1
Warning: multiple batches of metrics for period 1664217410000 2022-09-26T18:36:50.000Z
--------------------------------------
Metrics for period to: 18:37:10(+0000) (width: 0s)
--------------------------------------
vusers.created: ................................................................ 1
vusers.created_by_name.HomePage: ............................................... 1
Warning: multiple batches of metrics for period 1664217420000 2022-09-26T18:37:00.000Z
--------------------------------------
Metrics for period to: 18:37:20(+0000) (width: 0s)
--------------------------------------
vusers.created: ................................................................ 1
vusers.created_by_name.PlpPage: ................................................ 1
Warning: multiple batches of metrics for period 1664217430000 2022-09-26T18:37:10.000Z

**Comment by juanig1** [comment]
Hey, created [this](https://github.com/artilleryio/artillery/issues/1695) issue for further investigation. Been hitting this too occasionally.
Can't share an ETA yet, but will try to find some time to tackle it.
If you have a reproducible setup that's consistently missing metrics, please share!

**Comment by prateekbhaskarliquiloans** [comment]
This issue still persists.
